FHEM/lib/SD_Protocols.pm

```diff
@@ -590,12 +590,21 @@ sub mcBit2Hideki {
   my $id        = shift // carp 'protocol ID must be provided' && return (0,'no protocolId provided');
   my $mcbitnum  = shift // length $bitData;
 
-  my $message_start = index($bitData,'10101110');
+  if ($mcbitnum == 89) {                                   # optimization when the beginning was missing
+    my $bit0 = substr($bitData,0,1);
+    $bit0 = $bit0 ^ 1;
+    $bitData = $bit0 . $bitData;
+    $self->_logging( qq[lib/mcBit2Hideki, L=$mcbitnum add bit $bit0 at begin $bitData], 5 );
+  }
+
+  my $message_start = index($bitData,'10101110');         # normal rawMSG
   my $invert = 0;
+  my $message_start_invert = index($bitData,'01010001');  # invert rawMSG
+  # 10101110 can occur again in raw MSG -> comparison with inverted start 01010001
 
-  if ($message_start < 0) {
-    $bitData =~ tr/01/10/;                  # invert message
-    $message_start = index($bitData,'10101110');      # 0x75 but in reverse order
+  if ( $message_start < 0 || ( $message_start_invert!= -1 && $message_start > 0 && ($message_start_invert < $message_start) ) ) {
```

Comment:
Hast Du solche Datenssätze, also $bitdata?

Reply:
Hier solltest du fündig werden :-) Nicht wundern ich habe zur verdeutlichung kurzzeitig Hilfsausgaben ergänzt nur lokal bei mir.
Alle Daten im JSON außer MC;LL=-1018;LH=914;SL=-539;SH=436;D=A8E4F45AEDF40AF97595766;C=484;L=91;R=7;
weil diese nur im comment verankert ist :-)

```
1.) RAW von https://github.com/RFD-FHEM/RFFHEM/issues/944 welche ohne Anpassung nicht verarbeitet wurde
https://github.com/RFD-FHEM/SIGNALduino_TOOL/blob/6cd7ef2f3eb5cb9ad8cac28979a2b1099506b889/FHEM/lib/SD_Device_ProtocolList.json#L455-L461

2021.05.26 22:33:49 4: sduino_dummy: get rawmsg: MC;LL=-1018;LH=914;SL=-539;SH=436;D=A8E4F45AEDF40AF97595766;C=484;L=91;R=7;
2021.05.26 22:33:49 4: sduino_dummy: Parse_MC, Found manchester protocol id 10 clock 484 RSSI = -70.5 -> Oregon Scientific v2|v3
2021.05.26 22:33:49 4: sduino_dummy: Parse_MC, Found manchester protocol id 12 clock 484 RSSI = -70.5 -> Hideki
2021.05.26 22:33:49 4: sduino_dummy: lib/mcBit2Hideki, bitData               10101000111001001111010001011010111011011111010000001010111110010111010110010101011101100110
2021.05.26 22:33:49 4: sduino_dummy: lib/mcBit2Hideki, message_start        (10101110) -> pos 28
2021.05.26 22:33:49 4: sduino_dummy: lib/mcBit2Hideki, message_start_invert (01010001) -> pos 1
2021.05.26 22:33:50 4: sduino_dummy: lib/mcBit2Hideki, receive data is inverted
2021.05.26 22:33:50 4: sduino_dummy: lib/mcBit2Hideki, protocol converted to hex: 7536BA8A82BFC151AB6401 with 91 bits, messagestart 1
2021.05.26 22:33:50 4: sduino_dummy Hideki_Parse: incomming P12#7536BA8A82BFC151AB6401
2021.05.26 22:33:50 4: sduino_dummy Hideki crcCheck ok: cs1/cs2 0/100, rawData=7536BA8A82BFC151AB6401, count+2=9, length=11
2021.05.26 22:33:50 4: sduino_dummy Hideki_Parse: raw=7536BA8A82BFC151AB6401, decoded=755ace9e86c143f3fd6401
2021.05.26 22:33:50 4: sduino_dummy Hideki_Parse: SensorTyp = 30 decodedString = 755ace9e86c143f3fd6401

2.) RAW 1 aus JSON welche als Test der Funktion dient
https://github.com/RFD-FHEM/SIGNALduino_TOOL/blob/6cd7ef2f3eb5cb9ad8cac28979a2b1099506b889/FHEM/lib/SD_Device_ProtocolList.json#L444-L451
2021.05.26 22:36:10 4: sduino_dummy: get rawmsg: MC;LL=-1010;LH=938;SL=-519;SH=457;D=A8C2345ADD8E0B99F586F9C;C=487;L=90;R=57;
2021.05.26 22:36:10 4: sduino_dummy: Parse_MC, Found manchester protocol id 10 clock 487 RSSI = -45.5 -> Oregon Scientific v2|v3
2021.05.26 22:36:10 4: sduino_dummy: Parse_MC, Found manchester protocol id 12 clock 487 RSSI = -45.5 -> Hideki
2021.05.26 22:36:10 4: sduino_dummy: lib/mcBit2Hideki, bitData               10101000110000100011010001011010110111011000111000001011100110011111010110000110111110011100
2021.05.26 22:36:10 4: sduino_dummy: lib/mcBit2Hideki, message_start        (10101110) -> pos -1
2021.05.26 22:36:10 4: sduino_dummy: lib/mcBit2Hideki, message_start_invert (01010001) -> pos 1
2021.05.26 22:36:11 4: sduino_dummy: lib/mcBit2Hideki, receive data is inverted
2021.05.26 22:36:11 4: sduino_dummy: lib/mcBit2Hideki, protocol converted to hex: 75EFBA4A72BECC504F1801 with 91 bits, messagestart 1
2021.05.26 22:36:11 4: sduino_dummy Hideki_Parse: incomming P12#75EFBA4A72BECC504F1801
2021.05.26 22:36:11 4: sduino_dummy Hideki crcCheck ok: cs1/cs2 0/24, rawData=75EFBA4A72BECC504F1801, count+2=9, length=11
2021.05.26 22:36:11 4: sduino_dummy Hideki_Parse: raw=75EFBA4A72BECC504F1801, decoded=7531cede96c254f0d11801
2021.05.26 22:36:11 4: sduino_dummy Hideki_Parse: SensorTyp = 30 decodedString = 7531cede96c254f0d11801

3.) RAW 2 aus JSON welche als Test der Funktion dient
https://github.com/RFD-FHEM/SIGNALduino_TOOL/blob/6cd7ef2f3eb5cb9ad8cac28979a2b1099506b889/FHEM/lib/SD_Device_ProtocolList.json#L465-L472
2021.05.26 22:36:50 4: sduino_dummy: get rawmsg: MC;LL=-1022;LH=927;SL=-531;SH=442;D=A8EA345AD8BE0A3F6AA3F94;C=486;L=90;R=37;
2021.05.26 22:36:50 4: sduino_dummy: Parse_MC, Found manchester protocol id 10 clock 486 RSSI = -55.5 -> Oregon Scientific v2|v3
2021.05.26 22:36:50 4: sduino_dummy: Parse_MC, Found manchester protocol id 12 clock 486 RSSI = -55.5 -> Hideki
2021.05.26 22:36:50 4: sduino_dummy: lib/mcBit2Hideki, bitData               10101000111010100011010001011010110110001011111000001010001111110110101010100011111110010100
2021.05.26 22:36:50 4: sduino_dummy: lib/mcBit2Hideki, message_start        (10101110) -> pos -1
2021.05.26 22:36:50 4: sduino_dummy: lib/mcBit2Hideki, message_start_invert (01010001) -> pos 1
2021.05.26 22:36:51 4: sduino_dummy: lib/mcBit2Hideki, receive data is inverted
2021.05.26 22:36:51 4: sduino_dummy: lib/mcBit2Hideki, protocol converted to hex: 75EABA4A17BE07A91D5801 with 91 bits, messagestart 1
2021.05.26 22:36:51 4: sduino_dummy Hideki_Parse: incomming P12#75EABA4A17BE07A91D5801
2021.05.26 22:36:51 4: sduino_dummy Hideki crcCheck ok: cs1/cs2 0/88, rawData=75EABA4A17BE07A91D5801, count+2=9, length=11
2021.05.26 22:36:51 4: sduino_dummy Hideki_Parse: raw=75EABA4A17BE07A91D5801, decoded=753ecede39c209fb275801
2021.05.26 22:36:51 4: sduino_dummy Hideki_Parse: SensorTyp = 30 decodedString = 753ecede39c209fb275801
```

Reply:
Dauert leider noch, bin am Wochenende nicht dazu gekommen

Reply:
Geht in Ordnung.
Ich habe den Branch schonmal vorab geupdatet um die Konflikte zu beheben.

Reply:
Test ist ergänzt:)